sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/GetCatalogTests.cs

```diff
@@ -0,0 +1,281 @@
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForAVS()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("AVS", "eastus");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+            testForCatalogResult(catalogResult, "AVS", null, true, "eastus", "East US");
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForDataFactory()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("DataFactory", "eastus");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+            testForCatalogResult(catalogResult, "DataFactory", null, true, "eastus", "East US");
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForNetAppStorage()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("NetAppStorage", "westus");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+
+            Assert.NotNull(catalogResult);
+            Assert.IsTrue(catalogResult.Count > 0);
+            testForCatalogResult(catalogResult, "NetAppStorage", null, true, "westus", "West US");
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForAzureFiles()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("AzureFiles", "westus");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+            testForCatalogResult(catalogResult, "AzureFiles", null, true, "westus", "West US");
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForSqlEdge()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("SqlEdge");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+            testForCatalogResult(catalogResult, "SqlEdge");
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForVirtualMachineSoftware()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("VirtualMachineSoftware", publisherId: "test_test_pmc2pc1", offerId: "mnk_vmri_test_001", planId: "testplan001");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+            testForCatalogResult(catalogResult, "VirtualMachineSoftware");
+        }
+
+        private void testForCatalogResult(List<ReservationCatalog> catalogResult, string resourceTypeName, string alternateResourceTypeName = null, bool hasLocation = false, string location = null, string locationDisplayName = null)
```

Comment:
Considering naming convention and consistency, I'd like to rename this method as follows:
```suggestion
        private void TestGetCatalogResponse(List<ReservationCatalog> catalogResult, string resourceTypeName, string alternateResourceTypeName = null, bool hasLocation = false, string location = null, string locationDisplayName = null)
```